### ISSUE TITLE
Mixin InjectionError for Redirector useModifiedEnchantmentLevel (Forge 1.19.2)

### ISSUE THREAD

**Issue by MetaflameDragon** [MerchantPug/apugli#48]
We're using a medium modpack (79 mods) together with Origins, Origins++, and now Apugli.

Attempting to load the game with the latest version of Apugli (2.5.0) causes a crash on load, while 2.4.5 loads correctly. (However, with that version we ran into crashes caused by #45).

I managed to create a patch for us for the time being by removing `common.EnchantmentConfigurationMixin` from `apugli.forge.mixins.json`. Obviously suboptimal, but it allowed the game to start up.

I've yet to do testing by gradually enabling mods within our modpack, to see if this is caused by a mod conflict. I can try to do so if needed.

The exception stack trace is as follows:

```
Caught exception during event RegisterEvent dispatch for modid apoli
org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:250)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.service.modlauncher.MixinTransformationHandler.processClass(MixinTransformationHandler.java:131)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.launch.MixinLaunchPluginLegacy.processClass(MixinLaunchPluginLegacy.java:131)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.serviceapi.ILaunchPluginService.processClassWithFlags(ILaunchPluginService.java:156)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:88)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.TransformingClassLoader.maybeTransformClassBytes(TransformingClassLoader.java:50)
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.readerToClass(ModuleClassLoader.java:113)
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.lambda$findClass$15(ModuleClassLoader.java:219)
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.loadFromModule(ModuleClassLoader.java:229)
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.findClass(ModuleClassLoader.java:219)
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:135)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at TRANSFORMER/apoli@1.19.2-2.6.1.0/io.github.edwinmindcraft.apoli.common.condition.entity.EnchantmentCondition.<init>(EnchantmentCondition.java:11)
	at TRANSFORMER/forge@43.2.10/net.minecraftforge.registries.DeferredRegister.lambda$addEntries$1(DeferredRegister.java:388)
	at TRANSFORMER/forge@43.2.10/net.minecraftforge.registries.RegisterEvent.register(RegisterEvent.java:59)
	at TRANSFORMER/forge@43.2.10/net.minecraftforge.registries.DeferredRegister.addEntries(DeferredRegister.java:388)
	at TRANSFORMER/forge@43.2.10/net.minecraftforge.registries.DeferredRegister$EventDispatcher.handleEvent(DeferredRegister.java:330)
	at TRANSFORMER/forge@43.2.10/net.minecraftforge.registries.__EventDispatcher_handleEvent_RegisterEvent.invoke(.dynamic)
	at MC-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:73)
	at MC-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.EventBus.post(EventBus.java:315)
	at MC-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.EventBus.post(EventBus.java:296)
	at LAYER PLUGIN/javafmllanguage@1.19.2-43.2.10/net.minecraftforge.fml.javafmlmod.FMLModContainer.acceptEvent(FMLModContainer.java:107)
	at LAYER PLUGIN/fmlcore@1.19.2-43.2.10/net.minecraftforge.fml.ModLoader.lambda$postEventWithWrapInModOrder$35(ModLoader.java:315)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at LAYER PLUGIN/fmlcore@1.19.2-43.2.10/net.minecraftforge.fml.ModList.forEachModInOrder(ModList.java:225)
	at LAYER PLUGIN/fmlcore@1.19.2-43.2.10/net.minecraftforge.fml.ModLoader.postEventWithWrapInModOrder(ModLoader.java:313)
	at TRANSFORMER/forge@43.2.10/net.minecraftforge.registries.GameData.postRegisterEvents(GameData.java:341)
	at TRANSFORMER/forge@43.2.10/net.minecraftforge.common.ForgeStatesProvider.lambda$new$4(ForgeStatesProvider.java:25)
	at LAYER PLUGIN/fmlcore@1.19.2-43.2.10/net.minecraftforge.fml.ModLoader.lambda$dispatchAndHandleError$23(ModLoader.java:207)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at LAYER PLUGIN/fmlcore@1.19.2-43.2.10/net.minecraftforge.fml.ModLoader.dispatchAndHandleError(ModLoader.java:207)
	at LAYER PLUGIN/fmlcore@1.19.2-43.2.10/net.minecraftforge.fml.ModLoader.lambda$gatherAndInitializeMods$14(ModLoader.java:185)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at LAYER PLUGIN/fmlcore@1.19.2-43.2.10/net.minecraftforge.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:185)
	at TRANSFORMER/forge@43.2.10/net.minecraftforge.client.loading.ClientModLoader.lambda$begin$1(ClientModLoader.java:91)
	at TRANSFORMER/forge@43.2.10/net.minecraftforge.client.loading.ClientModLoader.lambda$createRunnableWithCatch$4(ClientModLoader.java:111)
	at TRANSFORMER/forge@43.2.10/net.minecraftforge.client.loading.ClientModLoader.begin(ClientModLoader.java:91)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.client.Minecraft.<init>(Minecraft.java:468)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.client.main.Main.m_239872_(Main.java:176)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.client.main.Main.main(Main.java:51)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at MC-BOOTSTRAP/fmlloader@1.19.2-43.2.10/net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$launchService$0(CommonClientLaunchHandler.java:27)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.Launcher.run(Launcher.java:106)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.Launcher.main(Launcher.java:77)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.8/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23)
	at cpw.mods.bootstraplauncher@1.1.2/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141)
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Redirector useModifiedEnchantmentLevel(Lnet/minecraft/world/item/enchantment/Enchantment;Lnet/minecraft/world/item/ItemStack;)I in apugli.forge.mixins.json:common.EnchantmentConfigurationMixin failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap apugli.refmap.json
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1362)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1051)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:325)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:383)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:365)
	at MC-BOOTSTRAP/org.spongepowered.mixin/org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	... 54 more
```
(This block is repeated one more time, the only change being on line 17 with entity being replaced by item: `at TRANSFORMER/apoli@1.19.2-2.6.1.0/io.github.edwinmindcraft.apoli.common.condition.entity.EnchantmentCondition.<init>(EnchantmentCondition.java:11)` -> `at TRANSFORMER/apoli@1.19.2-2.6.1.0/io.github.edwinmindcraft.apoli.common.condition.item.EnchantmentCondition.<init>(EnchantmentCondition.java:12)`)
Full crash log: https://drive.google.com/file/d/16ohvm-MF0LCd8T-pYK_FTEYHebk4swX6/view?usp=sharing

Mod list (Curseforge-compatible, contains a HTML with links otherwise): https://drive.google.com/file/d/1rLJnCZBSoZoDqOb0DsOEU1C5JYFyCspK/view?usp=sharing
I included my "patched" version, as well as a `.jar.disabled` Apugli 2.5.0

**Comment by ChrysanthCow** [comment]
I'll look into this soonish.

**Comment by ChrysanthCow** [comment]
This shouldn't be a compatibility issue, I think that it's just a broken version.

**Comment by ChrysanthCow** [comment]
Will be fixed in 2.5.1
